.github/workflows/smoke-tests.yml

```diff
@@ -0,0 +1,105 @@
+name: "System Smoke Tests"
+on:
+  push:
+    # run on pushes to master or release/*
+    branches:    
+      - master
+      - release/*
+  pull_request:
+    # run on pull requests targeting master
+    branches:
+      - master
+jobs:
+  setup:
+    runs-on: self-hosted
+    steps:
+    - name: Check Test Cluster
+      run: |
+        if kubectl get nodes; then
+          kubectl get nodes
+        else
+          echo "test cluster not active. Attempting to create through kind..."
+          kind delete cluster 2>/dev/null
+          kind create cluster
+          kubectl get nodes
+        fi
+  test-source:
+    runs-on: self-hosted
+    needs: setup
+    steps:
+    - uses: actions/checkout@v2
+    - name: Clean Cluster
+      run: |
+        kubectl delete all --all --wait=true --force --grace-period=0
+        sleep 5
+    - name: Deploy From Source
+      run: |
+        skaffold run
+    - name: Wait For Pods
+      run: |
+        sleep 60
+        kubectl get pods
+        echo waiting for ready state...
+        kubectl wait --for=condition=ready pod -l app=adservice --timeout=500s
```

Comment:
If kubectl wait has an option to wait on `deployment` rather than `pod` (I think it does), that would be preferable.

Reply:
The reason I say that is because no pods will be created if there's something more fundamental failing. So that might succeed.

However, Deployment will be there as it's part of `apply`, so it's better to check its readiness.

Reply:
That seems to work fine too. Done